### PR TITLE
Authentication project: Members Only

### PR DESCRIPTION
**Pull request by LOctavio**
For this project, we used the Devise gem to build an application where:

- Users can see posts
- Authenticated users can see posts and author of each post

### CODE REVIEW

**Review by Forison — COMMENTED** [review]
## Status:: Change required❌ 
# 🙋‍♂️ Team
* [ ] The routes should be limited to ```resources :posts, only: [:new, :create, :index]``` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_routes_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
- Currently, you have extra routes which are not needed per the requirement of this project.

* [ ] It uses `before_action` to restrict access to the `:new` and `:create` methods (uses ApplicationController helper method) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/before_action_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
- This will enforce that a user should be signed in before they can create a post
###### `new`
* [ ] it initializes an empty post `@post = Post.new` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_new_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
### That is it, kindly make the changes and submit the project for re-review😎.
#### Happy coding👨🏾‍💻 
#### Addo Forison

**Comment by nurislam-ab** [comment]
> ## Status:: Change required❌
> # 🙋‍♂️ Team
> * [ ]  The routes should be limited to `resources :posts, only: [:new, :create, :index]` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_routes_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
> 
> * Currently, you have extra routes which are not needed per the requirement of this project.
> 

okay we will do that



> * [ ]  It uses `before_action` to restrict access to the `:new` and `:create` methods (uses ApplicationController helper method) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/before_action_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))

not clear, need more details about this part http://joxi.ru/ZrJoznBtMBzGVm. what kind of helper methods should be created in the ApplicationController?


> * This will enforce that a user should be signed in before they can create a post
> 
but it is already done https://monosnap.com/file/wuJ7qDfduXNewuvcLucY6KRS2bIKrT

if the user clicks the **new post**, they are asked to log in firstly 

following the instructions the before_action was added at the top of the PostController 
http://joxi.ru/8AnnE8aU7JGMVA - odin 
http://joxi.ru/Dr87RwaUz8WbwA - code


> ###### `new`
> * [ ]  it initializes an empty post `@post = Post.new` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_new_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
> 
not clear, why we need that empty instance of the post?

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      else
+        format.html { render :new }
+        format.json { render json: @post.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  def index
+    @posts = Post.all
+    @post = Post.new
```
Comment (mcrd25) [inline app/controllers/posts_controller.rb]: this is usually found in #new action [OPTIONAL]

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
+class PostsController < ApplicationController
+  def new
+    @post = current_user.posts.build
```
Comment (mcrd25) [inline app/controllers/posts_controller.rb]: better to say '@post = Post.new`

Comment (mcrd25) [reply app/controllers/posts_controller.rb]: You're doing the current_user.posts.build in create, this is not necessary. `@post = Post.new` is simply to initialise the @post variable type

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -1,2 +1,3 @@
 class ApplicationController < ActionController::Base
+  before_action :authenticate_user!, only: %i[new create]
```
Comment (mcrd25) [inline app/controllers/application_controller.rb]: This should be in `PostController` See this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/before_action_requirement.png) form odin.

**Review by mcrd25 — CHANGES_REQUESTED** [review]
# Changes Required 🔍 

Hi Team! 👋 

Great job!
One thing though! You did not address the changes the previous reviewer requested correctly. Hopefully, you can understand now what was requested and why.

These are minor and you should be able to get the 👍  after! 😄 

#### Required changes
- See review comments not marked by [OPTIONAL]

***Please makes changes and submit another review request.
If you have any questions, let me know here or on slack and I will be glad to clarify with you. You can also reach me and connect with me on my social media or email.***

*Happy coding 😄,*
**[--Maya](https://github.com/mcrd25)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/) [GitLab](https://gitlab.com/mcrd25)

---
*P.S. all requested changes not marked by [OPTIONAL] or are not positive comments are **REQUIRED***

**Review by Ispirett — APPROVED** [review]
## Status Approved 🥇 

Great job folks with all changes made and completing this project 👍  You can now move on.

**Pull request merged**